packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ISharedObject } from "@fluidframework/shared-object-base/internal";
+import {
+	type ImplicitFieldSchema,
+	type InsertableTreeFieldFromImplicitField,
+	type ITree,
+	SharedTree,
+	type TreeView,
+	type TreeViewConfiguration,
+} from "@fluidframework/tree/internal";
+
+import { PureDataObject } from "./pureDataObject.js";
+
+/**
+ * Channel ID of {@link TreeDataObject}'s root {@link @fluidframework/tree#SharedTree}.
+ */
+const treeChannelId = "root";
+
+/**
+ * {@link @fluidframework/tree#SharedTree}-backed {@link PureDataObject | data object}.
+ *
+ * @internal
+ */
+export abstract class TreeDataObject<
+	TSchema extends ImplicitFieldSchema = ImplicitFieldSchema,
+> extends PureDataObject {
+	/**
+	 * The configuration used to initialize new documents, as well as to interpret (schematize) existing ones.
+	 *
+	 * @remarks
+	 * The fact that a single view schema is provided here (on the data object) makes it impossible to try and apply multiple different schema.
+	 * Since the view schema currently does not provide any adapters for handling differences between view and stored schema,
+	 * it's also impossible for this single view schema to handle multiple different stored schema.
+	 * Therefore, with this current API, two different applications (or different versions of the same application)
+	 * with differing stored schema requirements (as implied by their view schema) can not collaborate on the same tree.
+	 * The only schema evolution that's currently possible is upgrading the schema to one that supports a superset of what the old schema allowed,
+	 * and collaborating between clients which have view schema that exactly correspond to that stored schema.
+	 * Future work on tree as well as these utilities should address this limitation.
+	 */
+	public abstract readonly config: TreeViewConfiguration<TSchema>;
+
+	/**
+	 * The underlying tree.
+	 */
+	#tree: ITree | undefined;
+
+	/**
+	 * View of the underlying tree.
+	 */
+	#treeView: TreeView<TSchema> | undefined;
+
+	/**
+	 * Gets the underlying {@link @fluidframework/tree#ITree}.
+	 * @throws If the tree has not yet been initialized, this will throw an error.
+	 */
+	private get initializedTree(): ITree {
```

Comment:
i also think if we move all initialization to initializeInternal we can get rid of this, and the underlying `#tree` member.